lib/floe/workflow_base.rb

```diff
@@ -77,6 +77,11 @@ def output(context)
       context.output.to_json if end?(context)
     end
 
+    # overrides ValidationMixin#workflow_state?
+    def workflow_state?(field_value, _workflow)
+      @states_by_name ? @states_by_name.key?(field_value) : @payload["States"].include?(field_value)
```

Comment:
Hm if we need to fallback to `@payload` anyway we should probably just use that

Reply:
yeah isn't this the same?
```suggestion
      @payload["States"].include?(field_value)
```